client/rest-high-level/src/main/java/org/elasticsearch/client/ml/MlInfoResponse.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.ml;
+
+import org.elasticsearch.client.Validatable;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+
+public class MlInfoResponse implements Validatable {
+    private final Map<String, Object> info;
```

Comment:
are these well known values? Are they subject to change? Should this be more than just a generic map? cc @droberts195 cuz im not sure.

Reply:
Ahh, I just read `configurations where certain fields are missing because the end user was happy with the default value.` in the docs. So is it possible that `defaults` and `limits` will not both be in the map? If these are always here (as it looks like from the tests / docs in this PR), then maybe we should at least have 2 maps?

Reply:
We deliberately left things hazy about what would be in the output because it might change over time and we didn't want to get into BWC problems for this endpoint.

The transport client already returns a single map with all the values in so the way this endpoint is currently implemented matches what any transport client users would see.

I think the average end user would never use this endpoint because if they're happy with a default value in one of their configs they'd just have the value unset in their config.  If they want to know what those defaults are they can check the docs.

I think it's mainly UI implementors who might be interested in what the default values are via an API.  If someone outside the company ever decides to implement an ML config UI they'll probably complain about the haziness of this endpoint but I'd rather tie down the exact output format at that time (which may never come) rather than now.

Reply:
sounds like this is exactly what we want then! :shipit: 	!!!!